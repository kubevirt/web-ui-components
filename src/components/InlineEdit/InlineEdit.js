import React from 'react';
import PropTypes from 'prop-types';

import { InlineFormFactory } from '../Form/FormFactory';
import { Loading } from '../Loading/Loading';

export const InlineEdit = ({
  updating,
  editing,
  formFields,
  LoadingComponent,
  children,
  onFormChange,
  fieldsValues,
}) => {
  if (updating) {
    return <LoadingComponent />;
  }
  if (editing) {
    return <InlineFormFactory fields={formFields} fieldsValues={fieldsValues} onFormChange={onFormChange} />;
  }
  return <div>{children}</div>;
};

InlineEdit.propTypes = {
  updating: PropTypes.bool,
  editing: PropTypes.bool,
  formFields: PropTypes.object,
  LoadingComponent: PropTypes.func,
  children: PropTypes.node.isRequired,
  onFormChange: PropTypes.func,
  fieldsValues: PropTypes.object,
};

InlineEdit.defaultProps = {
  updating: false,
  editing: false,
  formFields: {},
  LoadingComponent: Loading,
  onFormChange: () => {},
  fieldsValues: {},
};
