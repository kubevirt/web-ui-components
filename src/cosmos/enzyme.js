import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
