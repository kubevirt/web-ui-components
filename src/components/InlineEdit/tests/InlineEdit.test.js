import React from 'react';
import { shallow } from 'enzyme';

import { InlineEdit } from '..';

import { default as InlineEditFixture } from '../fixtures/InlineEdit.fixture';

const testInlineEdit = () => <InlineEdit {...InlineEditFixture.props} />;

describe('<InlineEdit />', () => {
  it('renders correctly', () => {
    const component = shallow(testInlineEdit());
    expect(component).toMatchSnapshot();
  });
});
