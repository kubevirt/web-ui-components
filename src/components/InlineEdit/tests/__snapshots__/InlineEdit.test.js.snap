// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineEdit /> renders correctly 1`] = `
<div>
  inline edit
</div>
`;
